### ISSUE TITLE
Assertion failed 'found' during 'Profile instrumentation' 

### ISSUE THREAD

**Issue by kunalspathak** [dotnet/runtime#65987]
```c#
// Found by Antigen
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;
public class TestClass
{
    public struct S1
    {
    }
    private static List<string> toPrint = new List<string>();
    public void Method0()
    {
        unchecked
        {
            S1 s1_143 = new S1();
            S1 s1_144 = s1_143;
            Log("s1_144", s1_144);
            return;
        }
    }
    public static void Main(string[] args)
    {
        new TestClass().Method0();
        PrintLog();
    }
    [MethodImpl(MethodImplOptions.NoInlining)]
    public static void Log(string varName, object varValue)
    {
        toPrint.Add($"{varName}={varValue}");
    }
    public static void PrintLog()
    {
        foreach (var entry in toPrint)
        {
            Console.WriteLine(entry);
        }
    }
}
/*

Environment:

set COMPlus_TC_PartialCompilation=1
set COMPlus_TieredCompilation=1
set COMPlus_TieredPGO=1
set COMPlus_JitStress=1

Assert failure(PID 68104 [0x00010a08], Thread: 120288 [0x1d5e0]): Assertion failed 'found' in 'System.Text.OSEncoder:GetBytes(long,int,long,int,bool):int:this' during 'Profile instrumentation' (IL size 211)
    File: D:\git\dotnet-runtime\src\coreclr\jit\fgprofile.cpp Line: 1378
    Image: D:\git\dotnet-runtime\artifacts\tests\coreclr\windows.x64.Checked\tests\Core_Root\CoreRun.exe
*/

```

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

**Comment by AndyAyersMS** [comment]
This was an issue with using sparse edge profiles in a partially compiled method. We need to use block profiles.

Fixed by #66061.
